checker/src/main/java/alluxio/checker/MapReduceIntegrationChecker.java

```diff
@@ -0,0 +1,285 @@
+ * It will check whether Alluxio classes and Alluxio filesystem can be recognized
+ * in Hadoop task nodes.
+ */
+public class MapReduceIntegrationChecker {
+  private static final Logger LOG = LoggerFactory.getLogger(MapReduceIntegrationChecker.class);
+  private static final String FAIL_TO_FIND_CLASS_MESSAGE = "Please distribute "
+      + "the Alluxio client jar on the classpath of the application across different nodes.\n\n"
+      + "For details, please refer to: "
+      + "https://www.alluxio.org/docs/master/en/Running-Hadoop-MapReduce-on-Alluxio.html\n";
+  private static final String FAIL_TO_FIND_FS_MESSAGE = "Please check the fs.alluxio.impl "
+      + "and fs.AbstractFileSystem.alluxio.impl properties "
+      + "in core-site.xml file of your Hadoop installation.\n\n"
+      + "For details, please refer to: "
+      + "https://www.alluxio.org/docs/master/en/Running-Hadoop-MapReduce-on-Alluxio.html\n";
+  private static final String TEST_FAILED_MESSAGE = "***** Integration test failed. *****\n";
+  private static final String TEST_PASSED_MESSAGE = "***** Integration test passed. *****\n";
+
+  private Path mInputFilePath;
+  private Path mOutputFilePath;
+  private FileSystem mFileSystem;
+
+  /**
+   * In each mapper node, we will check whether this node can recognize
+   * Alluxio classes and filesystem.
+   */
+  protected static class CheckStatusMapper extends Mapper<Object, Text, Text, Text> {
+    /**
+     * Records the Status and IP address of each mapper task node.
+     */
+    protected void map(Object key, Text value, Context context)
+        throws IOException, InterruptedException {
+      context.write(new Text(CommonCheckerUtils.performIntegrationChecks().toString()),
+          new Text(CommonCheckerUtils.getAddress()));
+    }
+  }
+
+  /**
+   * In each reducer node, we will combine the IP addresses that have the same Status.
+   */
+  protected static class MergeStatusReducer extends Reducer<Text, Text, Text, Text> {
+    private Text mMergedAddress = new Text();
+
+    /**
+     * Merges the IP addresses of same Status.
+     */
+    protected void reduce(Text key, Iterable<Text> values, Context context)
+        throws IOException, InterruptedException {
+      String mergedAddress = "";
+      for (Text val : values) {
+        String nodeAddress = val.toString();
+        if (!mergedAddress.contains(nodeAddress)) {
+          mergedAddress = mergedAddress + " " + nodeAddress;
+        }
+      }
+      mMergedAddress.set(mergedAddress);
+      context.write(key, mMergedAddress);
+    }
+  }
+
+  /**
+   * Creates the HDFS filesystem to generate input and output files.
+   *
+   * @param conf Hadoop configuration
+   */
+  private void createHdfsFilesystem(Configuration conf) throws Exception {
+    // Inits HDFS File System Object
+    String systemUserName = System.getProperty("user.name");
+    System.setProperty("HADOOP_USER_NAME", systemUserName);
```

Comment:
I use the system username as the HADOOP_USER_NAME.  It works in local computer and AWS, but I am not sure how companies define their HADOOP_USER_NAME.  Do we need to add an argument that called HADOOP_USER_NAME to let users pass in their username?

Reply:
Sounds good! I will modify this, thanks!